File: JavaProgramming/src/main/java/special/solid_principle/composition/Composition.java
package special.solid_principle.composition;

import special.solid_principle.data.Address;
import special.solid_principle.data.User;

public class Composition {
    public static void main(String[] args) {
        User user1 = new User(1234, "Algokelvin", new Address(123, "SoloJakarta"));
        User user2 = new User(5412, "Jenda", new Address(342, "Jakarta"));

        System.out.println(user1.getName()+"-"+user1.getAddress().getLocation());
        System.out.println(user2.getName()+"-"+user2.getAddress().getLocation());
    }
}

File: KotlinProgramming/src/SOLID_CaseStudy/composition/Composition.kt
package SOLID_CaseStudy.composition

import SOLID_CaseStudy.modeldata.Address
import SOLID_CaseStudy.modeldata.User

fun main() {
    val user = User(
        id = 1234,
        name = "Algokelvin",
        address = Address(123, "Bandung")
    )
    println("Name: " + user.name)
    println("Address: " + user.address?.location)
}